pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -130,17 +128,46 @@ func (d *DependenciesDistributor) NeedLeaderElection() bool {
 
 func (d *DependenciesDistributor) discoverResources(period time.Duration) {
 	wait.Until(func() {
-		for _, gvr := range supportedTypes {
-			if d.InformerManager.IsHandlerExist(gvr, d.EventHandler) {
-				continue
+		newResources := lifted.GetDeletableResources(d.DiscoveryClientSet)
+		for gvr := range newResources {
+			if !d.InformerManager.IsHandlerExist(gvr, d.EventHandler) && !d.gvrDisabled(gvr) {
+				klog.Infof("Setup informer for %s", gvr.String())
+				d.InformerManager.ForResource(gvr, d.EventHandler)
 			}
-			klog.Infof("Setup informer for %s", gvr.String())
-			d.InformerManager.ForResource(gvr, d.EventHandler)
 		}
 		d.InformerManager.Start()
 	}, period, d.stopCh)
 }
 
+// gvrDisabled returns whether GroupVersionResource is disabled.
+func (d *DependenciesDistributor) gvrDisabled(gvr schema.GroupVersionResource) bool {
+	if d.SkippedResourceConfig == nil {
+		return false
+	}
+
+	if d.SkippedResourceConfig.GroupVersionDisabled(gvr.GroupVersion()) {
+		return true
+	}
+	if d.SkippedResourceConfig.GroupDisabled(gvr.Group) {
+		return true
+	}
+
+	gvks, err := d.RESTMapper.KindsFor(gvr)
+	if err != nil {
+		klog.Errorf("gvr(%s) transform failed: %v", gvr.String(), err)
+		return false
+	}
+	for _, gvk := range gvks {
+		if d.SkippedResourceConfig.GroupVersionKindDisabled(gvk) {
+			return true
+		}
+		if !d.ResourceInterpreter.HookEnabled(gvk, configv1alpha1.InterpreterOperationInterpretDependency) {
```

Comment:
I do not get it. Why this code is required?
```
if !d.ResourceInterpreter.HookEnabled(gvk, configv1alpha1.InterpreterOperationInterpretDependency) {
```

Reply:
updated.